### PR TITLE
#1025 heads

### PR DESCRIPTION
**Pull request by nezamj03**
## Changes

1. added heads role
2. changed file name to include all user utils
3. changed all instanced of user.role === X to use helpers
4. added HEAD seed data
5. changed some testing data roles
6. added some custom exceptions specific to making role exceptions (messages are easier too)

## Notes

did not add tests because essentially no role restrictions actually changed (only one instance but we test an extreme anyway)

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ x] All commits are tagged with the ticket number
- [ x] No linting errors / newline at end of file warnings
- [ x] All code follows repository-configured prettier formatting
- [ x] No merge conflicts
- [ x] All checks passing
- [ x] Screenshots of UI changes (see Screenshots section)
- [ x] Remove any non-applicable sections of this template
- [ x] Assign the PR to yourself
- [ x] No `yarn.lock` changes (unless dependencies have changed)
- [ x] Request reviewers & ping on Slack
- [ x] PR is linked to the ticket (fill in the closes line below)

Closes # 1025

### CODE REVIEW

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const ProposedSolutionsList: React.FC<ProposedSolutionsListProps> = ({ proposedS
           <ProposedSolutionView key={i} proposedSolution={proposedSolution} crReviewed={crReviewed} />
         ))}
       </div>
-      {crReviewed === undefined && auth.user?.role !== 'GUEST' ? (
+      {crReviewed === undefined && (auth.user ? !isGuest(auth.user.role) : false) ? (
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx:55]: what if we just made all these functions possibly take in undefined and if it's undefined it returns false or whatever

File: src/backend/src/utils/errors.utils.ts [hunk 2]
```diff
@@ -38,6 +38,36 @@ export class AccessDeniedException extends HttpException {
   }
 }
 
+export class AccessDeniedAdminException extends AccessDeniedException {
```
Comment (anthonybernardi) [inline src/backend/src/utils/errors.utils.ts:41]: I think calling this `AccessDeniedAdminOnlyException` would make more sense and be in line with the other ones

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
pretty epic ngl

File: src/shared/src/user-utils.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+export const isHead = (role: Role) => {
+  return isAtLeastRank(role, 4);
+};
+
+export const isLeadership = (role: Role) => {
+  return isAtLeastRank(role, 3);
+};
+
+export const isNotLeadership = (role: Role) => {
+  return isAtMostRank(role, 2);
```
[the remark this answers is not in the document]
Comment (nezamj03) [reply src/shared/src/user-utils.ts:41]: clever

**Review by anthonybernardi — APPROVED** [review]
W

**Review by anthonybernardi — APPROVED** [review]

**Pull request merged**